lib/active_storage_support/support_for_base64.rb

```diff
@@ -8,24 +8,48 @@ module SupportForBase64
       def has_one_base64_attached(name, dependent: :purge_later)
         has_one_attached name, dependent: dependent
 
-        add_helper_method(ActiveStorageSupport::Base64One, name, dependent: dependent)
+        generated_association_methods.class_eval <<-CODE, __FILE__, __LINE__ + 1
+          def #{name}
+            @active_storage_attached_#{name} ||= ActiveStorageSupport::Base64One.new("#{name}", self)
+          end
+          def #{name}=(attachable)
+            attachment_changes["#{name}"] =
+              if attachable.nil?
+                ActiveStorage::Attached::Changes::DeleteOne.new("#{name}", self)
+              else
+                ActiveStorage::Attached::Changes::CreateOne.new(
+                  "#{name}", self, ActiveStorageSupport::Base64One.from_base64(attachable)
+                )
+              end
+          end
+        CODE
       end
 
       def has_many_base64_attached(name, dependent: :purge_later)
         has_many_attached name, dependent: dependent
 
-        add_helper_method(ActiveStorageSupport::Base64Many, name, dependent: dependent)
-      end
-
-      def add_helper_method(type, name, dependent:)
-        class_eval <<-CODE, __FILE__, __LINE__ + 1
+        generated_association_methods.class_eval <<-CODE, __FILE__, __LINE__ + 1
           def #{name}
-            @active_storage_attached_#{name} ||=
-              #{type}.new("#{name}", self, dependent: #{dependent == :purge_later ? ':purge_later' : 'false'})
+            @active_storage_attached_#{name} ||= ActiveStorageSupport::Base64Many.new("#{name}", self)
           end
-
-          def #{name}=(data)
-            #{name}.attach(data)
+          def #{name}=(attachables)
+            if ActiveStorage.replace_on_assign_to_many
+              attachment_changes["#{name}"] =
+                if attachables.nil? || Array(attachables).none?
+                  ActiveStorage::Attached::Changes::DeleteMany.new("#{name}", self)
+                else
+                  ActiveStorage::Attached::Changes::CreateMany.new(
+                    "#{name}", self, ActiveStorageSupport::Base64Many.from_base64(attachables)
+                  )
+                end
+            else
+              if !attachables.nil? || Array(attachables).any?
```

Comment:
I think this makes no sense, I'll make a PR to rails to fix this but this should be an `&&`

Reply:
https://github.com/rails/rails/pull/37018